pkg/beholder/message.go

```diff
@@ -0,0 +1,257 @@
+package beholder
+
+import (
+	"fmt"
+
+	"github.com/go-playground/validator/v10"
+	"go.opentelemetry.io/otel/attribute"
+	otellog "go.opentelemetry.io/otel/log"
+)
+
+type Message struct {
+	Attrs Attributes
+	Body  []byte
+}
+
+type Metadata struct {
+	//	REQUIRED FIELDS
+	// Schema Registry URI to fetch schema
+	BeholderDataSchema string `validate:"required,uri"`
+
+	// OPTIONAL FIELDS
+	// The version of the CL node.
+	NodeVersion string
+	// mTLS public key for the node operator. This is used as an identity key but with the added benefit of being able to provide signatures.
+	NodeCsaKey string
+	// Signature from CSA private key.
+	NodeCsaSignature string
+	DonID            string
+	// The RDD network name the CL node is operating with.
+	NetworkName          []string
+	WorkflowID           string
+	WorkflowName         string
+	WorkflowOwnerAddress string
+	// Hash of the workflow spec.
+	WorkflowSpecID string
+	// The unique execution of a workflow.
+	WorkflowExecutionID string
+	// The address for the contract.
+	CapabilityContractAddress string
+	CapabilityID              string
+	CapabilityVersion         string
+	CapabilityName            string
+	NetworkChainID            string
+}
+
+func (m Metadata) Attributes() Attributes {
+	return Attributes{
+		"node_version":                m.NodeVersion,
+		"node_csa_key":                m.NodeCsaKey,
+		"node_csa_signature":          m.NodeCsaSignature,
+		"don_id":                      m.DonID,
+		"network_name":                m.NetworkName,
+		"workflow_id":                 m.WorkflowID,
+		"workflow_name":               m.WorkflowName,
+		"workflow_owner_address":      m.WorkflowOwnerAddress,
+		"workflow_spec_id":            m.WorkflowSpecID,
+		"workflow_execution_id":       m.WorkflowExecutionID,
+		"beholder_data_schema":        m.BeholderDataSchema,
+		"capability_contract_address": m.CapabilityContractAddress,
+		"capability_id":               m.CapabilityID,
+		"capability_version":          m.CapabilityVersion,
+		"capability_name":             m.CapabilityName,
+		"network_chain_id":            m.NetworkChainID,
+	}
+}
+
+type Attributes map[string]any
+
+func NewAttributes(attrKVs ...any) Attributes {
+	attrs := make(Attributes, len(attrKVs)/2)
+	attrs.Add(attrKVs...)
+	return attrs
+}
+
+func (a Attributes) Add(attrKVs ...any) Attributes {
```

Comment:
I think some direct unit testing of this method would be helpful for the reader.

Reply:
Removed it https://github.com/smartcontractkit/chainlink-common/commit/4e80921db29d6f144890789656570af35a17f528